test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadEventTest/VThreadEventTest.java

```diff
@@ -0,0 +1,180 @@
+    private static boolean failed;
+    private static List<Thread> threads = new ArrayList(TCNT1);
+
+    public static void main(String[] args) throws Exception {
+        if (Runtime.getRuntime().availableProcessors() < 8) {
+            log("WARNING: test expects at least 8 processors.");
+        }
+        Counter ready1 = new Counter(THREAD_CNT);
+        Counter ready2 = new Counter(THREAD_CNT);
+        Counter mready = new Counter(1);
+
+        try (ExecutorService executorService = Executors.newVirtualThreadPerTaskExecutor()) {
+            for (int tCnt = 0; tCnt < TCNT1; tCnt++) {
+                executorService.execute(() -> {
+                    synchronized (threads) {
+                        threads.add(Thread.currentThread());
+                    }
+                    log("test1 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    try {
+                        // timeout is big enough to keep mounted untill interrupted
+                        Thread.sleep(20000);
+                    } catch (InterruptedException ex) {
+                        // it is expected, ignore
+                    }
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            for (int tCnt = 0; tCnt < TCNT2; tCnt++) {
+                executorService.execute(() -> {
+                    log("test2 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    while (!attached) {
+                        // keep mounted
+                    }
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            for (int tCnt = 0; tCnt < TCNT3; tCnt++) {
+                executorService.execute(() -> {
+                    log("test3 vthread started");
+                    ready1.decr();
+                    mready.await();
+                    while (!attached) {
+                        // keep mounted
+                    }
+                    LockSupport.parkNanos(10 * TIMEOUT_BASE);
+                    ready2.decr();
+                    completedNo++;
+                });
+            }
+            ready1.await();
+            mready.decr();
+            VirtualMachine vm = VirtualMachine.attach(String.valueOf(ProcessHandle.current().pid()));
+            vm.loadAgentLibrary("VThreadEventTest");
+            Thread.sleep(100);
+            log("main: completedNo: " + completedNo);
+            attached = true;
+            for (Thread t : threads) {
+                t.interrupt();
+            }
+            ready2.await();
+        }
+        // wait not more than 10 secs until all VirtualThreadEnd events are sent
+        for (int sleepNo = 0; sleepNo < 10 && threadEndCount() < THREAD_CNT; sleepNo++) {
+            log("main: wait iter: " + sleepNo);
+            Thread.sleep(100);
```

Comment:
sleep(1000)? (comment before the loop tells about 10 secs)

Reply:
Good catch. Leonid suggested to make waiting with unlimited sleeps, so the test timeout will work.
Made it like below:
```
        // wait until all VirtualThreadEnd events are sent
        while (int sleepNo = 1; threadEndCount() < THREAD_CNT; sleepNo++) {
            if (sleepNo % 100 == 0) { // 10 sec period of waiting
                log("main: wait iter: " + sleepNo);
            }
```